marimo/_ast/sql_visitor.py

```diff
@@ -315,109 +313,52 @@ def find_sql_refs(
     Returns:
         A list of table and schema names referenced in the statement.
     """
-    if not DependencyManager.duckdb.has():
+
+    # Use sqlglot to parse ast (https://github.com/tobymao/sqlglot/blob/main/posts/ast_primer.md)
+    if not DependencyManager.sqlglot.has():
         return []
 
-    import duckdb
+    from sqlglot import exp, parse
+    from sqlglot.optimizer.scope import build_scope
 
-    tokens = duckdb.tokenize(sql_statement)
-    token_extractor = TokenExtractor(
-        sql_statement=sql_statement, tokens=tokens
-    )
     refs: list[str] = []
-    cte_names: set[str] = set()
-    i = 0
 
-    # First pass - collect CTE names
-    while i < len(tokens):
-        if token_extractor.is_keyword(i, "with"):
-            i += 1
-            # Handle optional parenthesis after WITH
-            if token_extractor.token_str(i) == "(":
-                i += 1
-            while i < len(tokens):
-                if token_extractor.is_keyword(i, "select"):
-                    break
-                if (
-                    token_extractor.token_str(i) == ","
-                    or token_extractor.token_str(i) == "("
-                ):
-                    i += 1
-                    continue
-                cte_name = token_extractor.strip_quotes(
-                    token_extractor.token_str(i)
+    def append_refs_from_table(table: exp.Table) -> None:
+        if table.catalog == "memory":
+            # Default in-memory catalog, only include table name
+            refs.append(table.name)
+        else:
+            # We skip schema if there is a catalog
+            # Because it may be called "public" or "main" across all catalogs
+            # and they aren't referenced in the code
+            if table.catalog:
+                refs.append(table.catalog)
+            elif table.db:
+                refs.append(table.db)  # schema
+
+            if table.name:
+                refs.append(table.name)
+
+    expression_list = parse(sql_statement)
```

Comment:
Since we only support `duckdb` right now, and since duckdb has some special functions/syntax that other dialects don't have, I think this should be:

```suggestion
    expression_list = parse(sql_statement, dialect="duckdb")
```